src/phases/recall-phase.ts

```diff
@@ -31,13 +31,27 @@ export class RecallPhase extends PokemonPhase {
 
   public override start(): void {
     const pokemon = this.getPokemonAtFieldIndex();
-    if (!pokemon?.isOnField()) {
-      this.end();
+    if (!pokemon?.isActive(true) || !pokemon.isOnField()) {
+      super.end();
       return;
     }
     this.pokemon = pokemon;
 
-    this.recall().then(() => this.end());
+    const { currentBattle, phaseManager } = globalScene;
+    const { turnManager } = currentBattle;
+    if (this.switchType === SwitchType.SWITCH && turnManager.tryPursueTarget(pokemon)) {
+      // Reschedule this phase for after the scheduled Pursuit is fully resolved
+      // TODO: this might cause issues if Pursuit KOs the target
+      phaseManager.unshiftPhase(this);
+      super.end();
+      return;
+    }
+
+    this.recall()
+      .then(() => this.end())
+      .catch(() => {
+        throw new Error("RecallPhase: Recall process failed unexpectedly");
+      });
```

Comment:
Can you make `RecallPhase#start` async and then do this?
```ts
await this.recall();
this.end();
```

If not, then just add a todo instead of useless code.
```suggestion
    // biome-ignore lint/nursery/noFloatingPromises: TODO: refactor async code
    this.recall().then(() => this.end());
```